xarray/core/weighted.py

```diff
@@ -308,6 +467,18 @@ def std(
             self._weighted_std, dim=dim, skipna=skipna, keep_attrs=keep_attrs
         )
 
+    def quantile(
+        self,
+        q,
+        dim: Optional[Union[Hashable, Iterable[Hashable]]] = None,
```

Comment:
```suggestion
        q,
        *,
        dim: Optional[Union[Hashable, Iterable[Hashable]]] = None,
```
I prefer as few positional arguments as possible, but maybe it would be annoying when switching to non-weighted?